prow/git/git.go

```diff
@@ -288,9 +296,17 @@ func (r *Repo) Push(repo, branch string) error {
 		return errors.New("cannot push without credentials - configure your git client")
 	}
 	r.logger.Infof("Pushing to '%s/%s (branch: %s)'.", r.user, repo, branch)
-	remote := fmt.Sprintf("https://%s:%s@%s/%s/%s", r.user, r.pass, github, r.user, repo)
+
+	gitURL, err := url.Parse(r.base)
+	if err != nil {
+		return err
```

Comment:
`Push()` can fail in three ways:
* User not provided
* Base fails to parse
* Push command failed

I would prefer we do something like `fmt.Errorf("parse base: %v", err)` to help distinguish things here rather than just pass the err through in each scenario

Reply:
Added a more descriptive error message.